com.ca.lsp.cobol/lsp-service-cobol/src/test/java/com/ca/lsp/cobol/service/InitializeConfigurationTest.java

```diff
@@ -0,0 +1,86 @@
+/*
+ * Copyright (c) 2020 Broadcom.
+ * The term "Broadcom" refers to Broadcom Inc. and/or its subsidiaries.
+ *
+ * This program and the accompanying materials are made
+ * available under the terms of the Eclipse Public License 2.0
+ * which is available at https://www.eclipse.org/legal/epl-2.0/
+ *
+ * SPDX-License-Identifier: EPL-2.0
+ *
+ * Contributors:
+ *    Broadcom, Inc. - initial API and implementation
+ */
+
+package com.ca.lsp.cobol.service;
+
+import com.ca.lsp.cobol.model.ConfigurationSettingsStorable;
+import com.ca.lsp.cobol.service.providers.ClientProvider;
+import com.ca.lsp.cobol.service.providers.SettingsProvider;
+import com.ca.lsp.cobol.utils.Utility;
+import com.google.gson.JsonArray;
+import com.google.gson.JsonObject;
+import org.eclipse.lsp4j.ConfigurationParams;
+import org.eclipse.lsp4j.InitializedParams;
+import org.eclipse.lsp4j.services.LanguageClient;
+import org.junit.Test;
+import org.mockito.Mockito;
+
+import java.util.ArrayList;
+import java.util.List;
+import java.util.concurrent.CompletableFuture;
+import java.util.concurrent.ExecutionException;
+
+import static org.mockito.Mockito.*;
+
+/**
+ * This test class checks if the workspace calls client configuration at the initialize method and
+ * retrieves the setting configuration
+ */
+public class InitializeConfigurationTest {
+  private LanguageClient client = Mockito.mock(LanguageClient.class);
+  private ClientProvider provider = new ClientProvider();
+  private SettingsProvider settingsProvider = Mockito.mock(SettingsProvider.class);
```

Comment:
Due to you don't share these fields between different methods, I think it is better to keep them local, but it is up to you